### ISSUE TITLE
How to show bare translation keys 

### ISSUE THREAD

**Issue by webdevbe** [i18next/i18next#947]
Hi,

Is it possible to show only the keys without the translation? What I'm trying to accomplish is that if a url contains a specific fragment, the translation keys are shown so the user is able to identify the key. Didn't find something in the docs immediately.

Thanks

**Comment by jamuhl** [comment]
https://www.i18next.com/faq.html#process

How to handle with changes in e2e tests?

For e2e tests a good tactic is to set language to `cimode` on init. This will set i18next to always return the key on calling i18next.t. Want to add the namespace to returned value change appendNamespaceToCIMode: true on init.

=> if you use the language detector just open your page with querystring param `?lng=cimode`

**Comment by webdevbe** [comment]
Hmm yes that could be an option however, in SPA's you'll usually setup i18next only once. However when routing to any page, the parameter will disappear but the lng parameter still be set to cimode, causing the page to still show the bare keys. Any thoughts or do I need to check on every page if this param is still in the url and if not, call i18nexts reloadResources method?

**Comment by jamuhl** [comment]
no need to call reloadResources (is this for react?) - just set the language to the detected language - or reload the page -> cimode does not get cached so won't be there after reload.

if you need this for your translators you might consider: https://www.youtube.com/watch?v=JrqZK918Bhg&feature=youtu.be 

(full length vid: https://www.youtube.com/watch?v=9NOzJhgmyQE&feature=youtu.be )

**Comment by webdevbe** [comment]
Well I rather don't reload the page in a single page application but what about setting the language to another language. Does that mean that from wherever I call i18next.setLng('fr') it will translated the page without refreshing?

**Comment by jamuhl** [comment]
depends...if using react-i18next yes...if something else...no

reload or not reload...really depends on the usecase...showing the keys is not part of the customer experience i guess...but something needed for your translators...idk...depends on why you need to show the keys....

**Comment by jamuhl** [comment]
closing for now...feel free to reopen
